src/EntityFramework6.Npgsql/SqlGenerators/SqlBaseGenerator.cs

```diff
@@ -829,6 +829,23 @@ protected string GetDbType(EdmType edmType)
 
         public override VisitedExpression Visit([NotNull] DbCaseExpression expression)
         {
+            // Check for COALESCE like CASE
+            if (expression.When.Count == 1 &&
+                expression.When[0].ExpressionKind == DbExpressionKind.IsNull)
+            {
+                var is_null = expression.When[0] as DbIsNullExpression;
+                if (is_null.Argument.Equals(expression.Else))
+                {
```

Comment:
Minor nitpick: code naming conventions, is_null should be isNull. To be more consistent with code below, it should be isNullExpression.